org.eclipse.m2e.pde.ui/src/org/eclipse/m2e/pde/ui/target/editor/internal/viewer/TableColumnSortListener.java

```diff
@@ -0,0 +1,45 @@
+/*******************************************************************************
+ * Copyright (c) 2022 Patrick Ziegler and others.
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v2.0
+ * which accompanies this distribution, and is available at
+ * https://www.eclipse.org/legal/epl-v20.html
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ *
+ * Contributors:
+ *      Patrick Ziegler - initial API and implementation
+ *******************************************************************************/
+package org.eclipse.m2e.pde.ui.target.editor.internal.viewer;
+
+import org.eclipse.jface.viewers.TableViewer;
+import org.eclipse.swt.SWT;
+import org.eclipse.swt.events.SelectionAdapter;
+import org.eclipse.swt.events.SelectionEvent;
+import org.eclipse.swt.widgets.TableColumn;
+
+/**
+ * <p>
+ * This class is used to control the sorting behavior of the Maven artifacts in
+ * the dependency editor. Artifacts can be sorted in ascending or descending
+ * order by clicking on the corresponding column header. A table can only be
+ * sorted by a single column at a time.
+ * </p>
+ */
+public class TableColumnSortListener extends SelectionAdapter {
+	private final TableViewer viewer;
+	private int sortDirection;
+
+	public TableColumnSortListener(TableViewer viewer) {
+		this.viewer = viewer;
+	}
+
+	@Override
+	public void widgetSelected(SelectionEvent e) {
+		sortDirection = sortDirection != SWT.UP ? SWT.UP : SWT.DOWN;
```

Comment:
sortDirection would probably better be a boolean and then map it to the integer afterwards? Using `!=` here makes this line looking more complex as it is.